code/modules/ninja/ninja_explosive.dm

```diff
@@ -33,5 +28,24 @@
 	if(!detonator)
 		return
 	var/datum/antagonist/ninja/ninja_antag = detonator.mind.has_antag_datum(/datum/antagonist/ninja)
+	//buuuuuuuuuuuuuuut the c4 may have moved
+	if(!check_loc(detonator, ninja_antag))
+		return
 	var/datum/objective/plant_explosive/objective = locate() in ninja_antag.objectives
 	objective.completed = TRUE
+
+/obj/item/grenade/c4/ninja/proc/check_loc(mob/user, datum/antagonist/ninja/ninja_antag)
+	var/datum/objective/plant_explosive/objective = locate() in ninja_antag.objectives
```

Comment:
Wouldn't this result in a runtime? I haven't messed with antag datums in a hell of a long time but it would be weird if has_antag_datum() returned the datum if the mind doesn't have it in the first place

Reply:
No, on line 16 it gets the datum, and then on 17 it immediately checks if it actually exists